src/infill.cpp

```diff
@@ -782,6 +783,44 @@ void Infill::generateLinearBasedInfill(
     }
 }
 
+void Infill::resolveIntersection(const coord_t at_distance, const Point& intersect, Point& connect_start, Point& connect_end, InfillLineSegment* a, InfillLineSegment* b)
+{
+    // Select wich ends of the line need to 'bend'.
+    const bool forward_line_a = a->end == connect_start;
+    const bool forward_line_b = b->start == connect_end;
+    auto& bend_a = forward_line_a ? a->end_bend : a->start_bend;
+    auto& bend_b = forward_line_b ? b->start_bend : b->end_bend;
+    auto& end_a = forward_line_a ? a->altered_end : a->altered_start;
+    auto& end_b = forward_line_b ? b->altered_start : b->altered_end;
+
+    // Initialize 'bends'.
+    assert(! bend_a.has_value());
+    assert(! bend_b.has_value());
+    bend_a.emplace(0, 0);
+    bend_b.emplace(0, 0);
```

Comment:
Oops, shouldn't have committed this one... As you can see just above this fragment, I assert if it doesn't have a value already. This never went off during debug for me. So this will always fail.

Secondly, I fill them in later. This will fail when the lines are parallel however, but I check beforehand if the original lines cross (which are not the same as the ones now crossed...), so I thought that should not happen -- which is maybe where this bug has its origin.